openwisp_radius/base/models.py

```diff
@@ -1163,9 +1165,19 @@ def clean(self):
         if self._state.adding and self.user.is_active:
             logger.warning(_(f'user {self.user} is already active'))
             raise ValidationError(_('This user is already active.'))
-        if not self.user.phone_number:
-            logger.warning(_(f'user {self.user} does not have a phone number'))
-            raise ValidationError(_('This user does not have a phone number.'))
+        another_user_has_phone_number = (
+            self._meta.model.objects.filter(phone_number=self.phone_number)
+            .exclude(user=self.user)
+            .exists()
+        )
```

Comment:
please do the following:

- create a test in which a user is created with a phone number, but no phone token is created for this user (the number may be entered manually by an administrator), then create another user which tries to change the phone number via the API to the same phone number of the first user, expect a validation error. The test should fail
- fix the test by using the user model for the lookup (load the model with django's `get_user_model()`)